youtube_dl/extractor/kth.py

```diff
@@ -0,0 +1,31 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import smuggle_url
+
+
+class KTHIE(InfoExtractor):
+    _VALID_URL = r'https?://play\.kth\.se/(?:.+/)+(?P<id>[a-z0-9_]+)'
```

Comment:
`.+/` might be rather greedy. Try `(?:[^/]+/)`.

Can the id use upper-case letters?

Reply:
The ID never contains any upper case letters.